autoemulate/core/plotting.py

```diff
@@ -412,8 +411,143 @@ def create_and_plot_slice(
+        output_dim==1.
+    """
+    if levels is None:
+        levels = np.linspace(0.01, 0.99, 49)
+    levels = np.asarray(levels)
+
+    samples = y_pred.sample((n_samples,))
+
+    # compute empirical intervals using sample quantiles
+    empirical_list = []
+    for p in levels:
+        lower_q = (1.0 - p) / 2.0
+        upper_q = 1.0 - lower_q
+        lower = torch.quantile(samples, float(lower_q), dim=0)
+        upper = torch.quantile(samples, float(upper_q), dim=0)
+        inside = (y_true >= lower) & (y_true <= upper)
+        if joint:
+            inside_all = inside.all(dim=-1)
+            empirical = inside_all.float().mean().item()
+        else:
+            # marginal per-dim coverage
+            empirical = inside.float().mean(dim=0).cpu().numpy()
+        empirical_list.append(empirical)
+
+    empirical_arr = np.asarray(empirical_list)
+
+    return levels, empirical_arr
+
+
+def plot_calibration_from_distributions(
+    y_pred: DistributionLike,
+    y_true: TensorLike,
+    levels: np.ndarray | None = None,
+    n_samples: int = 2000,
+    joint: bool = False,
+    title: str | None = None,
+    legend: bool = True,
+    figsize: tuple[int, int] | None = None,
+):
+    """Plot calibration curve(s) given predictive distributions and true values.
+
+    This draws empirical coverage (y-axis) against nominal coverage (x-axis).
+
+    Parameters
+    ----------
+    y_pred: DistributionLike
+        The emulator predicted distribution.
+    y_true: TensorLike
+        The true values.
+    levels: array-like, optional
+        Nominal coverage levels (between 0 and 1). If None, a default grid is
+        used.
+    n_samples: int
+        Number of Monte-Carlo samples to draw from the predictive
+        distribution to compute empirical intervals.
+    joint: bool
+        If True and the predictive outputs are multivariate, compute joint
+        coverage (i.e., the true vector must lie inside the interval for all
+        dimensions). If False (default), compute marginal coverage per output
+        dimension and return the mean across data points.
+    title: str | None
+        An optional title for the plot. Defaults to None (no title).
+    legend: bool
+        Whether to display a legend. Defaults to True.
+    figsize: tuple[int, int] | None
+        The size of the figure to create. If None, a default size is used.
+    """
+    levels, empirical = coverage_from_distributions(
+        y_pred, y_true, levels=levels, n_samples=n_samples, joint=joint
     )
+    print(empirical.shape)
+
+    if figsize is None:
+        figsize = (6, 6)
+    fig, ax = plt.subplots(figsize=figsize)
+
+    if empirical.shape[1] == 1:
+        ax.plot(levels, empirical, marker="o", label="empirical")
+    else:
+        # multiple outputs: plot each dimension
+        for i in range(empirical.shape[1]):
+            ax.plot(levels, empirical[:, i], marker="o", label=f"dim {i}")
```

Comment:
Perhaps we could indicate its for the output in the label, e.g.:
```suggestion
            ax.plot(levels, empirical[:, i], marker="o", label=f"$y_{i}$")
```